ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -848,8 +955,366 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
+    uint32_t weight_matrix_width = weights_shape.padded_shape()[3];
+    uint32_t weight_matrix_height_ntiles = weight_matrix_height / tt::constants::TILE_HEIGHT;
+    uint32_t weight_matrix_width_ntiles  =  weight_matrix_width / tt::constants::TILE_WIDTH;
+
+    uint32_t per_core_out_matrix_width_ntiles = tt::div_up(pconfig.per_core_out_matrix_width, tt::constants::TILE_WIDTH);
+    uint32_t per_core_out_matrix_height_ntiles = tt::div_up(pconfig.per_core_out_matrix_height, tt::constants::TILE_HEIGHT);
+
+    uint32_t num_blocks_act_h_per_core =
+        (per_core_out_matrix_height_ntiles + act_block_h_ntiles - 1) / act_block_h_ntiles;
+    uint32_t out_block_h_ntiles_padded = num_blocks_act_h_per_core * act_block_h_ntiles;
+
+    if(shard_layout == TensorMemoryLayout::WIDTH_SHARDED)
+    {
+
+        uint32_t conv_output_c_per_core = per_core_out_matrix_width_ntiles * tt::constants::TILE_WIDTH;
+
+        uint32_t output_size_per_core_in_bytes = per_core_out_matrix_width_ntiles *
+                                                 per_core_out_matrix_height_ntiles *
+                                                 tt::tile_size(datatype_to_dataformat_converter(output_dtype));
+
+        uint32_t act_block_num_bytes = act_block_num_tiles * input_tile_size;
+        uint32_t tilized_act_block_num_bytes = act_block_num_tiles * output_tile_size;
+
+
+        uint32_t weight_block_w_ntiles = per_core_out_matrix_width_ntiles;
+        uint32_t weight_block_num_tiles = weight_block_w_ntiles * act_block_w_ntiles; //act_block_w_ntiles == weight_block_h_ntiles
+        uint32_t weight_block_num_bytes = weight_block_num_tiles * weights_tile_size;
+
+        uint32_t bias_block_num_bytes = per_core_out_matrix_width_ntiles * bias_tile_size;
+
+        uint32_t out_block_num_tiles = per_core_out_matrix_height_ntiles * per_core_out_matrix_width_ntiles;
+
+        uint32_t num_blocks_act_w = weight_matrix_height_ntiles / act_block_w_ntiles;
+
+        packer_l1_acc = packer_l1_acc && ((enable_bias && num_blocks_act_w > 1) || (num_blocks_act_w > 2));
+
+        auto interm_dtype =
+            packer_l1_acc ? (fp32_dest_acc_en ? DataType::FLOAT32 : DataType::BFLOAT16) : output_dtype;
+
+        uint32_t partial_tile_size = tt::tile_size(datatype_to_dataformat_converter(interm_dtype));
+
+        uint32_t partials_block_num_bytes = out_block_num_tiles * partial_tile_size;
+
+
+        //CB 0
+        uint32_t cb0_size = tilized_act_block_num_bytes; tt::log_debug(tt::LogOp, "CB0 Size: {}", cb0_size);
+
+        //CB 1
+        uint32_t cb1_size = weight_block_num_bytes; tt::log_debug(tt::LogOp, "CB1 Size: {}", cb1_size);
+
+        //CB 2
+        uint32_t cb2_size = bias_block_num_bytes; tt::log_debug(tt::LogOp, "CB2 Size: {}", cb2_size);
+
+        //CB 5
+        uint32_t cb5_size = 64; tt::log_debug(tt::LogOp, "CB5 Size: {}", cb5_size);
+
+        //CB 6
+        uint32_t cb6_size = act_block_num_bytes; tt::log_debug(tt::LogOp, "CB6 Size: {}", cb6_size);
```

Comment:
cb6 cb27 is not very descriptive, can we put names on these (same names we use in factory?)

Reply:
Done.